### PR TITLE
Requested changes completed

### PR DESCRIPTION
**Pull request by kakhilreddy**

### CODE REVIEW

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/types/AddConceptTypeController.java [hunk 1]
```diff
@@ -21,22 +28,28 @@
...
     }
     
     @RequestMapping(value="/auth/concepts/types/add", method=RequestMethod.POST)
-    public String post(ConceptTypeForm form, Model model, Principal principal) {
-        
-        if (form.getName() != null && !form.getName().trim().isEmpty()) {
-            conceptTypeManager.create(form, userManager.findByUsername(principal.getName()));
-        } else {
+    public String post(@Validated @ModelAttribute("conceptTypeForm") ConceptTypeForm form, BindingResult result, Model model, Principal principal) {
+    	if (result.hasErrors()) {
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/types/AddConceptTypeController.java]: indentation

File: citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+        Mockito.when(conceptManager.save(Mockito.argThat(new ConceptArgMatcher(concept)))).thenReturn(concept);
+        Mockito.when(typeManager.save(Mockito.argThat(new ConceptTypeArgMatcher(type)))).thenReturn(type);
+        helperToTest.updateCitation(citation, form, user);
+        assertTrue(equalsCitation(citation, updatedCitation));
+    }
+
+    @Test
+    public void test_updateCitation_getByURIAndOwner() {
+
+        Mockito.when(conceptManager.getByUriAndOwner("www.google.com", user)).thenReturn(concept);
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java]: any string you use more than once, should be an constant.

File: citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+
+            if (!conceptArg.getUri().equals(conceptToBeTested.getUri()))
+                return false;
+
+            if (!conceptArg.getOwner().equals(conceptToBeTested.getOwner()))
+                return false;
+
+            return true;
+
+        }
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java]: please follow our code style

File: citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        dbConcept.setUri(URI_FORM_INPUT);
+    }
+
+    @Test
+    public void test_addConcept_Success() {
+
+        // return null for adding concept
+        Mockito.when(conceptManager.getByUriAndOwner(conceptForm.getUri(), user)).thenReturn(null);
+        citationConceptValidator.validate(conceptForm, errors);
+        assert (errors.getFieldError("uri") == null);
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java]: let's use `assertNull`

File: citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        assertTrue(!errors.hasErrors());
+
+    }
+
+    @Test
+    public void test_addConcept_Uri_Exists() {
+
+        Mockito.when(conceptManager.getByUriAndOwner(conceptForm.getUri(), user)).thenReturn(concept);
+        citationConceptValidator.validate(conceptForm, errors);
+        assert (errors.getFieldError("uri") != null);
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java]: `assertNotNull`

File: citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        conceptForm.setConceptId("CON1");
+        dbConcept.setOwner(user1);
+        Mockito.when(conceptManager.get("CON1")).thenReturn(dbConcept);
+        Mockito.when(conceptManager.getByUriAndOwner(conceptForm.getUri(), user)).thenReturn(concept);
+        citationConceptValidator.validate(conceptForm, errors);
+        assert (errors.getFieldError("uri") != null);
+
+    }
+
+}
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java]: any string that you use more than once needs to be a constant

File: citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/ConceptTypeValidatorTest.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        conceptTypeForm.setConceptTypeId("CTY1");
+        dbConceptType.setOwner(user1);
+        Mockito.when(conceptTypeManager.get("CTY1")).thenReturn(dbConceptType);
+        Mockito.when(conceptTypeManager.getByUriAndOwner(conceptTypeForm.getUri(), user)).thenReturn(conceptType);
+        conceptTypeValidator.validate(conceptTypeForm, errors);
+        assert (errors.getFieldError("uri") != null);
+
+    }
+
+}
```
Comment (jdamerow) [inline citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/ConceptTypeValidatorTest.java]: same as for the other test cases

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/types/AddConceptTypeController.java (hunk 1 above)
Comment (kakhilreddy) [reply citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/types/AddConceptTypeController.java]: I fixed it.
 I edited this file last month when I did not set the indentation properties.
Probably I missed it that time.

**Pull request closed without merging**